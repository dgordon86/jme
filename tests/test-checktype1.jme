main()
{
 int = 8;
 float = 8.8;
 str = "String";
 bool = true;
 vec = [1,2,3];
 matx = [1,2; 3,4];
 map = new| "a" => "a", "b" => "b", "c" => "c" |;
 
 print("Check Int");
 print(is_Int(int));
 print(is_Int(float));

 print("Check Float");
 print(is_Float(float));
 print(is_Float(int));
 
 print("Check String");
 print(is_String(str));
 print(is_String(int));
 
 print("Check Boolean");
 print(is_Bool(bool));
 print(is_Bool(str));
 
 print("Check Vector");
 print(is_Vector(vec));
 print(is_Vector(bool));
 
 print("Check Matrix");
 print(is_Matrix(matx));
 print(is_Matrix(vec));
 
 print("Check Map");
 print(is_Map(map));
 print(is_Map(matx));
 
}
