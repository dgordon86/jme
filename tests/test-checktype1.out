Check Int
true
false
Check Float
true
false
Check String
true
false
Check Boolean
true
false
Check Vector
true
false
Check Matrix
true
false
Check Map
true
false